pulsar-client-go/examples/reader/reader.go

```diff
@@ -0,0 +1,48 @@
+//
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+//
+
+package main
+
+import (
+	"../../pulsar"
+	"fmt"
+	"log"
+)
+
+func main() {
+	client := pulsar.NewClient().
+		ServiceUrl("pulsar://localhost:6650").Build()
+
+	reader, err := client.NewReader().Topic("my-topic").StartFromEarliest().Create()
```

Comment:
Any reason `StartFromEarliest()` is not a constant/enum like subscription type? I could imagine `-2` for earliest, `-1` for latest, and `>0` as the message id for the topic.

Reply:
the `MessageId` is actually an object and not just a int/long. I was try to specify a `const` pointing to such objects, but with no success..

Reply:
Ah right, they are objects. Yea `const` only works with primitive types. You can use pre-defined variables in this case and check identity.

```go
var (
  EarliestMessage MessageId = newMessageId()
  LatestMessage   MessageId = newMessageId()
)
reader := &Reader{
  StartMessage: EarliestMessage,
}

// Implementation..
switch reader.StartMessage {
case EarliestMessage:
  // ..
case LatestMessage:
  // ..
case nil:
  // ..
default:
  // user defined..
}
```